### DISCUSSION TITLE
Support for dragging .xlsx files outside sandbox to excel

### DISCUSSION THREAD

**Discussion by lmou523** [sandboxie-plus/Sandboxie#2955] [Various ideas and suggestions]
I found that if excel already had the file open, dragging and dropping the file into the table control would be ineffective.
drag-and-drop in the sandbox will work.However, dragging and dropping from inside the sandbox will work. Dragging files from outside the sandbox to the table control will not work.

**Comment by lmou523** [comment]
I debugged it and when the drag is outside the sandbox, calling __sys_GetPropW in Ole_DoDragDrop does not return NULL and then excel does not respond.
If it is a sandbox drag, __sys_GetPropW returns NULL. And then excel can open the file
And I noticed that their HWND are different

**Comment by lmou523** [comment]
Start excel in the sandbox and open another file
Sandbox out drag into excel
![image](https://github.com/sandboxie-plus/Sandboxie/assets/59944420/77ff9d88-d647-4411-a4f5-ff9695d2b72e)
Sandbox drag into excel
![image](https://github.com/sandboxie-plus/Sandboxie/assets/59944420/6b3ab270-9fbe-4377-a287-fbd6b9156255)
The sandbox drag is followed by a call to the original function
![image](https://github.com/sandboxie-plus/Sandboxie/assets/59944420/a1f9ae6f-1153-49e4-8fae-ab9343b455b9)

**Comment by lmou523** [comment]
Is this dragging through com? Can I just add an OpenClsid configuration? How do I determine its Clsid?
(Just conjecture)

**Reply by DavidXanatos** [reply]
I havn't looked into it yet.
You can open all Clsids by specifying a all 0's Clsid.
Or you can use the Open all Com option here  fortesting:
![grafik](https://github.com/sandboxie-plus/Sandboxie/assets/3890945/aa6db528-48ba-466b-90e3-8f0365efaa5c)

**Reply by isaak654** [reply]
> How do I determine its Clsid? (Just conjecture)

1. Add `ClsidTrace=*` to your sandbox, then use the [trace log](https://sandboxie-plus.github.io/sandboxie-docs/PlusContent/TraceLog.html) feature and check the output
2. Make sure to remove `ClsidTrace=*` after you're done
3. You can add any COM entry from the COM tab (as shared above)

**Reply by lmou523** [reply]
> I havn't looked into it yet. You can open all Clsids by specifying a all 0's Clsid. Or you can use the Open all Com option here fortesting: ![grafik](https://user-images.githubusercontent.com/3890945/240902565-aa6db528-48ba-466b-90e3-8f0365efaa5c.png)

Can you investigate the reason? I just made some attempts, but it didn't solve the problem.
This feature is very important to me, and I mainly want to know how excel works

**Reply by isaak654** [reply]
Did you try with [`OpenWinClass=*`](https://sandboxie-plus.github.io/sandboxie-docs/Content/OpenWinClass.html) ?

**Reply by lmou523** [reply]
> Did you try with [`OpenWinClass=*`](https://sandboxie-plus.github.io/sandboxie-docs/Content/OpenWinClass.html) ?

Yes, excel simply won't work, and I tried to find com.
I may have guessed wrong, excel doesn't rely on com

**Reply by lmou523** [reply]
> Did you try with [`OpenWinClass=*`](https://sandboxie-plus.github.io/sandboxie-docs/Content/OpenWinClass.html) ?

I also added OpenWinClass=EXCEL7, but that didn't work either

**Reply by lmou523** [reply]
> Did you try with [`OpenWinClass=*`](https://sandboxie-plus.github.io/sandboxie-docs/Content/OpenWinClass.html) ?

oh my god, excel seems to work when I add OpenWinClass=*

**Reply by isaak654** [reply]
> oh my god, excel seems to work when I add OpenWinClass=*

Because it's the most effective rule, alternatively you could use [`OpenWinClass=#`](https://sandboxie-plus.github.io/sandboxie-docs/Content/OpenWinClass.html).

I'm not sure why, but adding single window classes one by one from a proper Trace Log won't work.

There are various compatibility issues with the general wildcard rule, for example: 
https://forum.xanasoft.com/threads/how-to-open-links-from-other-programs-in-an-already-running-sandboxed-web-browser.585/

**Reply by isaak654** [reply]
The lead developer has pushed a fix for release v1.9.6 regarding the drag-and-drop issue with Excel's data grid: 9455e96a699cbc665f791e191f2a13bb40783ab0

Small caveaut: `UseDragDropHack=process.exe,n` may or may not work with other programs that have drag-and-drop issues in sandboxed windows, and it's essentially a temporary measure.

**Comment by nabbi070-maker** [comment]
[EL697332AD.xlsx](https://github.com/user-attachments/files/28452278/EL697332AD.xlsx)
قسم لي الملف الي شهرين في نفس السنة
